hs_core/models.py

```diff
@@ -736,30 +740,17 @@ def create(cls, **kwargs):
             raise ValidationError("Publisher element can't be created for a resource that "
                                   "is not yet published.")
 
-        publisher_CUAHSI = "Consortium of Universities for the Advancement of Hydrologic " \
-                           "Science, Inc. (CUAHSI)"
-
-        if resource.files.all():
-            # if the resource has content files, set CUAHSI as the publisher
-            if 'name' in kwargs:
-                if kwargs['name'].lower() != publisher_CUAHSI.lower():
-                    raise ValidationError("Invalid publisher name")
-
-            kwargs['name'] = publisher_CUAHSI
-            if 'url' in kwargs:
-                if kwargs['url'].lower() != 'https://www.cuahsi.org':
-                    raise ValidationError("Invalid publisher URL")
+        if 'name' in kwargs:
+            if kwargs['name'].lower() != cls.publisher_name_CUAHSI.lower():
+                raise ValidationError("Invalid publisher name")
+        else:
+            kwargs['name'] = cls.publisher_name_CUAHSI
 
-            kwargs['url'] = 'https://www.cuahsi.org'
+        if 'url' in kwargs:
+            if kwargs['url'].lower() != cls.publisher_url_CUAHSI.lower():
+                raise ValidationError("Invalid publisher URL")
         else:
-            # make sure we are not setting CUAHSI as publisher for a resource
```

Comment:
I think this also needs to be kept which allowed to specify a different publisher (non Hydroshare)  for resources that don't have content files (reference resources like ref time series resources).